src/boot/bootctl.c

```diff
@@ -123,12 +123,8 @@ static int verify_esp(
 
         errno = 0;
         b = blkid_new_probe_from_filename(t);
-        if (!b) {
-                if (errno == 0)
-                        return log_oom();
-
-                return log_error_errno(errno, "Failed to open file system \"%s\": %m", p);
-        }
+        if (!b)
+                return log_error_errno(errno ?: ENOMEM, "Failed to open file system \"%s\": %m", p);
```

Comment:
The alternative would be:
```diff
- log_error_errno(errno ?: ENOMEM, "...")
+ log_error_errno(errno > 0 ? errno : ENOMEM, "...")
```
This is much more verbose. I can prepare a patch if you think this is really necessary.

> still don't like it if we log errno constants we are making up instead of those we receive...

Hm, there are two cases. For ENOMEM, I'd say that having NULL returned is the same as errno==-ENOMEM, just in different form. For EIO, yeah, I can agree.